### ISSUE TITLE
Improve IDE0041 and IDE0084 analyzer

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn#48669]
**Brief description:**

Improve IDE0041 and IDE0084 to include other ways for equality/inequality check against null.

**Languages applicable:**

Both.

**Code example that the analyzer should report:**

```c#
class C
{
    public static void Main ()
    {
        var s = typeof (C);
        if ((object) s == null) // should suggest `s is null` instead
            return;
    }
}
```

**Comment by CyrusNajmabadi** [comment]
This is what is suggested:

![image](https://user-images.githubusercontent.com/4564579/96373579-4e96d480-1122-11eb-80d4-7652109013a3.png)
